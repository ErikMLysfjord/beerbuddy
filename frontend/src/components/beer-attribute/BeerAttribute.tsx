import styles from "./BeerAttribute.module.css";

export interface BeerAttributeProps {
  icon: string;
  altText: string;
  attribute: string;
  value: string | number | undefined;
}

/**
 * A component for displaying a beer attribute.
 * This is used to display more detailed information about a beer.
 * @param icon - The icon to display.
 * @param altText - The alt text for the icon.
 * @param attribute - The attribute heading.
 * @param value - The value of the attribute.
 * @returns - The beer attribute component.
 */
const BeerAttribute = ({
  attribute,
  value,
  icon,
  altText,
}: BeerAttributeProps) => {
  return (
    <div className={styles.attributeContainer} aria-label={attribute}>
      <div className={styles.attributeHeaderContainer}>
        <img height={"32px"} width={"20px"} src={icon} alt={altText} />
        <h3 className={styles.attributeHeading}>{attribute}</h3>
      </div>
      <p className={styles.attributeText}>{value}</p>
    </div>
  );
};
export default BeerAttribute;
