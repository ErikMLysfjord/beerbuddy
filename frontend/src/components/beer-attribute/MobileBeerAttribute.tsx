import styles from "./BeerAttribute.module.css";
import { BeerAttributeProps } from "./BeerAttribute";

interface MobileBeerAttributeProps {
  attributeProps: BeerAttributeProps[];
}

const MobileBeerAttribute = ({ attributeProps }: MobileBeerAttributeProps) => {
  return (
    <div className={styles.mobileContainer}>
      {attributeProps.map((prop) => {
        if (prop.value !== null) {
          return (
            <div>
              <div className={styles.mobileHeaderContainer}>
                <img
                  height={"32px"}
                  width={"20px"}
                  src={prop.icon}
                  alt={prop.altText}
                />
                <div className={styles.textContainer}>
                  <h3 className={styles.mobileHeading}>{prop.attribute}</h3>
                  <p className={styles.mobileText}>{prop.value}</p>
                </div>
              </div>
            </div>
          );
        }
      })}
    </div>
  );
};
export default MobileBeerAttribute;
