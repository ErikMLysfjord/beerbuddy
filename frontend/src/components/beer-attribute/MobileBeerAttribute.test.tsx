import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import MobileBeerAttribute from "./MobileBeerAttribute";
import { axe } from "jest-axe";
import { BeerAttributeProps } from "./BeerAttribute";

const attributeProps: BeerAttributeProps[] = [
  {
    attribute: "Testdata 1",
    altText: "alternative text",
    icon: "",
    value: 1,
  },
  {
    attribute: "Testdata 2",
    altText: "alternative text",
    icon: "",
    value: 2,
  },
  {
    attribute: "Testdata 3",
    altText: "alternative text",
    icon: "",
    value: 3,
  },
];

const Template = () => <MobileBeerAttribute attributeProps={attributeProps} />;

describe("MobileBeerAttribute", () => {
  it("is accessible", async () => {
    const { container } = render(<Template />);
    expect(await axe(container)).toHaveNoViolations();
  });

  it("renders correctly", () => {
    const { container } = render(<Template />);
    expect(container).toMatchSnapshot();
  });

  it("renders with correct values", () => {
    const { getByText, getAllByAltText } = render(<Template />);
    expect(getByText("Testdata 1")).toBeInTheDocument();
    expect(getByText("Testdata 2")).toBeInTheDocument();
    expect(getByText("Testdata 3")).toBeInTheDocument();
    expect(getAllByAltText("alternative text")).toHaveLength(3);
  });
});
