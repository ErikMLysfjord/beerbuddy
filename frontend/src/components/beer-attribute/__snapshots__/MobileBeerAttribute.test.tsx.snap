// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MobileBeerAttribute > renders correctly 1`] = `
<div>
  <div
    class="_mobileContainer_eeaf5b"
  >
    <div>
      <div
        class="_mobileHeaderContainer_eeaf5b"
      >
        <img
          alt="alternative text"
          height="32px"
          src=""
          width="20px"
        />
        <div
          class="_textContainer_eeaf5b"
        >
          <h3
            class="_mobileHeading_eeaf5b"
          >
            Testdata 1
          </h3>
          <p
            class="_mobileText_eeaf5b"
          >
            1
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        class="_mobileHeaderContainer_eeaf5b"
      >
        <img
          alt="alternative text"
          height="32px"
          src=""
          width="20px"
        />
        <div
          class="_textContainer_eeaf5b"
        >
          <h3
            class="_mobileHeading_eeaf5b"
          >
            Testdata 2
          </h3>
          <p
            class="_mobileText_eeaf5b"
          >
            2
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        class="_mobileHeaderContainer_eeaf5b"
      >
        <img
          alt="alternative text"
          height="32px"
          src=""
          width="20px"
        />
        <div
          class="_textContainer_eeaf5b"
        >
          <h3
            class="_mobileHeading_eeaf5b"
          >
            Testdata 3
          </h3>
          <p
            class="_mobileText_eeaf5b"
          >
            3
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
